pkg/frontegg/scim_groups.go

```diff
@@ -47,6 +64,149 @@ type SCIMGroupsResponse struct {
 	Groups []ScimGroup `json:"groups"`
 }
 
+// AddRolesToGroupParams represents the parameters for adding roles to a group.
+type AddRolesToGroupParams struct {
+	RoleIds []string `json:"roleIds"`
+}
+
+// CreateSCIMGroup creates a new group in Frontegg.
+func CreateSCIMGroup(ctx context.Context, client *clients.FronteggClient, params GroupCreateParams) (*ScimGroup, error) {
+	endpoint := fmt.Sprintf("%s%s", client.Endpoint, SCIMGroupsApiPathV1)
+	payloadBytes, err := json.Marshal(params)
+	if err != nil {
+		return nil, err
+	}
+	body := bytes.NewReader(payloadBytes)
+
+	req, err := http.NewRequestWithContext(ctx, "POST", endpoint, body)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+client.Token)
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := client.HTTPClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		var sb strings.Builder
+		_, err = io.Copy(&sb, resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("error creating SCIM group: status %d, response: %s", resp.StatusCode, sb.String())
+	}
+
+	var group ScimGroup
+	if err := json.NewDecoder(resp.Body).Decode(&group); err != nil {
+		return nil, err
+	}
+
+	return &group, nil
+}
+
+// UpdateSCIMGroup updates an existing group in Frontegg.
+func UpdateSCIMGroup(ctx context.Context, client *clients.FronteggClient, groupID string, params GroupUpdateParams) error {
+	endpoint := fmt.Sprintf("%s%s/%s", client.Endpoint, SCIMGroupsApiPathV1, groupID)
+	payloadBytes, err := json.Marshal(params)
+	if err != nil {
+		return err
+	}
+	body := bytes.NewReader(payloadBytes)
+
+	req, err := http.NewRequestWithContext(ctx, "PATCH", endpoint, body)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+client.Token)
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := client.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		var sb strings.Builder
+		_, err = io.Copy(&sb, resp.Body)
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("error updating SCIM group: status %d, response: %s", resp.StatusCode, sb.String())
+	}
+
+	return nil
+}
+
+// DeleteSCIMGroup deletes an existing group in Frontegg.
+func DeleteSCIMGroup(ctx context.Context, client *clients.FronteggClient, groupID string) error {
+	endpoint := fmt.Sprintf("%s%s/%s", client.Endpoint, SCIMGroupsApiPathV1, groupID)
+
+	req, err := http.NewRequestWithContext(ctx, "DELETE", endpoint, nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+client.Token)
+
+	resp, err := client.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		var sb strings.Builder
+		_, err = io.Copy(&sb, resp.Body)
```

Comment:
Might be worth extracting some helper functions for this stuff? Seems pretty repetitive.

Reply:
Good idea!

I've created a tracking issue to do that for the whole Frontegg client package: https://github.com/MaterializeInc/terraform-provider-materialize/issues/528